ledger_text_formats.cpp

```diff
@@ -1204,11 +1204,15 @@ std::string ledger_format
     ,std::pair<int,oenum_format_style> f
     )
 {
-    std::stringstream interpreter;
-    std::locale loc;
-    std::locale new_loc(loc, new comma_punct);
-    interpreter.imbue(new_loc);
-    interpreter.setf(std::ios_base::fixed, std::ios_base::floatfield);
+    static std::stringstream interpreter{[]() {
+        std::stringstream ss;
+        std::locale loc;
+        std::locale new_loc(loc, new comma_punct);
+        ss.imbue(new_loc);
+        ss.setf(std::ios_base::fixed, std::ios_base::floatfield);
+        return std::move(ss);
```

Comment:
This is a bug: you're returning a reference (so it's an rvalue reference -- but it's still a reference) to a local object which is undefined behaviour.

Just doing `return ss;` should be just as efficient because of NRVO, but, just to be sure, could you please recheck that it still results in the same 10%+ speed up? TIA!